### PR TITLE
Sa lifecycle

### PR DESCRIPTION
**Pull request by chris-pateman**
### Change description ###
correct issue with container loop
correct issue with policy loop
correct ReadMe example

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,4 +99,8 @@ sa_policy = [
       }
     }
   ]
+<<<<<<< HEAD
```
Comment (timja) [inline README.md:102]: merge conflicts

File: containers.tf [hunk 2]
```diff
@@ -1,8 +1,8 @@
 
 resource "azurerm_storage_container" "container" {
-  for_each = var.containers
+  count = length(var.containers)
```
Comment (timja) [inline containers.tf:3]: don't use count that means that ordering changes will cause re-creation use for_each

you may need to add a `toset` here

File: README.md (hunk 1 above)
Comment (chris-pateman) [reply README.md]: corrected

File: README.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ Currently there is only version deletion policy coded in, but it can be expanded
 ```
 sa_policy = [
     {
-      name = "Blob Retention Policy"
+      name = "BlobRetionPolicy"
```
[the remark this answers is not in the document]
Comment (chris-pateman) [reply README.md]: No I didn't :) corrected from merge

File: containers.tf (hunk 2 above)
Comment (chris-pateman) [reply containers.tf]: Couldn't use toset as doesn't support list of objects
Changed to use for_each

Comment (chris-pateman) [reply containers.tf]: Couldn't use toset as doesn't support list of objects
Changed to use for_each 

Tested with remote project in plan with containers

**Review by timja — APPROVED** [review]

**Pull request merged**